moved/src/state_actor.rs

```diff
@@ -225,11 +225,19 @@ impl<
 
     fn execute_transactions(
         &mut self,
-        transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
+        mut transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
         base_fee: U256,
     ) -> ExecutionOutcome {
         let mut total_tip = U256::ZERO;
         let mut outcomes = Vec::new();
+        let first_tx = transactions.next();
+        let l1_fee = match first_tx.as_ref() {
+            Some((_, NormalizedExtendedTxEnvelope::DepositedTx(deposit_tx), _)) => {
+                Some(self.l1_fee.for_deposit(deposit_tx.data.as_ref()))
+            }
+            _ => None,
+        };
+        let transactions = iter::once(first_tx).flatten().chain(transactions);
```

Comment:
nit: This line rebuilds the `transactions` I believe. Maybe we should leave a comment.